### DISCUSSION TITLE
Dev environment websocket errors

### DISCUSSION THREAD

**Discussion by Sampozzo** [mastodon/mastodon#21760] [Q&A]
Hello, i followed the info on the page https://docs.joinmastodon.org/dev/setup/ on windows 11 WSL Ubuntu 20.04 and all seems to work except the websocket:

i have this error in chrome console:

`index.js?ef83:53 WebSocket connection to 'ws://localhost:4000/api/v1/streaming/?' failed: 
open @ index.js?ef83:53
onBackoffReady @ index.js?ef83:68`

and this on the server:

```
23:47:08 stream.1  | WARN Starting worker 68
23:47:08 stream.1  | WARN Worker 68 now listening on 127.0.0.1:4000 
23:47:09 stream.1  | ERR! TypeError [ERR_INVALID_ARG_TYPE]: The "key" argument must be of type string or an instance of ArrayBuffer, Buffer, TypedArray, DataView, KeyObject, or CryptoKey. Received null 
23:47:09 stream.1  | WARN Worker 68 exiting 
```

Can i have some help on this? thank you very much!

**Comment by Marty-Crabneck** [comment]
This can also happen on a regular Mastodon server. With pg version 8.8.0 and a bunch of command line options for node I got at least a decent error message ("Uncaught Error: SASL: SCRAM-SERVER-FIRST-MESSAGE: client password must be a string").
`export DB_HOST; export DB_PORT; export DB_NAME; export DB_USER; export DB_PASS` at the end of .env.production fixed it.
HTH
